api/src/main/java/com/example/api/service/ranking/RankingService.java

```diff
@@ -0,0 +1,103 @@
+import com.example.api.service.activity.result.GraphTaskResultService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+@Transactional
+public class RankingService {
+    private final UserRepo userRepo;
+    private final GraphTaskResultRepo graphTaskResultRepo;
+    private final FileTaskResultRepo fileTaskResultRepo;
+    private final GraphTaskResultService graphTaskResultService;
+    private final ProfessorFeedbackService professorFeedbackService;
+
+    public List<RankingEntry> getRanking() {
+        return userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
+                .stream()
+                .map(this::studentToRankingEntry)
+                .sorted(Comparator.comparingDouble(RankingEntry::getPoints).reversed())
+                .collect(Collectors.toList());
+    }
+
+    public List<RankingEntry> getRankingForGroup(String groupName) {
+        return userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
+                .stream()
+                .filter(student -> Objects.equals(student.getGroup().getName(), groupName))
+                .map(this::studentToRankingEntry)
+                .sorted(Comparator.comparingDouble(RankingEntry::getPoints).reversed())
+                .collect(Collectors.toList());
+    }
+
+    public List<RankingEntry> getSearchedRanking(String search) {
+        String searchLower = search.toLowerCase();
+        return userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
+                .stream()
+                .filter(student ->
+                                student.getFirstName().toLowerCase().contains(searchLower) ||
+                                student.getLastName().toLowerCase().contains(searchLower) ||
+                                student.getHeroType().toString().toLowerCase().contains(searchLower) ||
+                                student.getGroup().getName().toLowerCase().contains(searchLower))
+                .map(this::studentToRankingEntry)
+                .sorted(Comparator.comparingDouble(RankingEntry::getPoints).reversed())
+                .collect(Collectors.toList());
+    }
+
+    private RankingEntry studentToRankingEntry(User student) {
+        RankingEntry rankingEntry = new RankingEntry(student);
+        rankingEntry.setPoints(getStudentPoints(student));
+        return rankingEntry;
+    }
+
+    private Double getGraphTaskPoints(User student) {
+        return graphTaskResultRepo.findAllByUser(student)
+                .stream()
+                .mapToDouble(task -> {
+                    try {
+                        return graphTaskResultService.getAndSetAllPoints(task.getId());
```

Comment:
Nie musisz brać puntków w taki sposób, po tym jak ktoś skończy ekspedycje to liczba uzyskanych punktów zapisuje mu się w polu receivedPoints - najlepiej bierz to po prostu stąd  (na dole widać, że dla każdego TaskResult mamy pointsReceived).

<img width="440" alt="Zrzut ekranu 2022-08-15 o 15 35 20" src="https://user-images.githubusercontent.com/73038429/184645576-9b763065-94c6-41a1-abb4-b8828ac4261e.png">